orm/src/main/java/org/hibernate/search/v6poc/entity/orm/model/impl/HibernateOrmPropertyModel.java

```diff
@@ -84,23 +89,31 @@ public String getName() {
 	@Override
 	public PropertyHandle getHandle() {
 		if ( handle == null ) {
-			handle = new GetterPropertyHandle( name, getter );
+			try {
+				handle = introspector.createPropertyHandle( name, member );
+			}
+			catch (IllegalAccessException | RuntimeException e) {
+				throw new SearchException( "Exception while retrieving property handle for '"
+						+ getName() + "' on '" + holderTypeModel + "'", e );
+			}
 		}
 		return handle;
 	}
 
 	Type getGetterGenericReturnType() {
-		Method method = getter.getMethod();
-		Member member = getter.getMember();
 		// Try to preserve generics information if possible
-		if ( method != null ) {
-			return method.getGenericReturnType();
+		if ( member instanceof Method ) {
+			return ( (Method) member ).getGenericReturnType();
 		}
-		else if ( member != null && member instanceof Field ) {
+		else if ( member instanceof Field ) {
 			return ( (Field) member ).getGenericType();
 		}
 		else {
-			return getter.getReturnType();
+			throw new AssertionFailure( "Unexpected type for a " + Member.class.getName() + ": " + member );
```

Comment:
This shouldn't happen but you don't really have the unexpected type information which is the useful one.

Reply:
Right, changed.